### DISCUSSION TITLE
solid-ssr/static rollup development configuration

### DISCUSSION THREAD

**Discussion by ghost** [solidjs/solid#1044] [Q&A]
Hi,

I have been looking into **solid-ssr/static** (https://github.com/solidjs/solid/tree/main/packages/solid-ssr) and to put it shortly, I do not know how to set up **rollup for development**. In the examples provided there is no dev script, only start or build. I was able to somehow make the solid-ssr/ssr working in the development mode but I had no luck with static option.

I tried to add for example: rollup-plugin-livereload, rollup-plugin-serve, @rollup/plugin-run. I do not know what I am doing wrong.

Could someone please tell me, show me, navigate me to the solution? 

Thank you,
Wojciech

**Comment by ryansolid** [comment]
I see yeah dev mode is one thing, but also solid-refresh is needed for HMR. We've only ever tested nollup so I don't know how those other plugins work.

I believe at least some of them will do full page reload without need for our plugin. So I'd start there. Unfortunately their default behavior should have nothing to do with what we are doing with Solid and simply reload the page on re-build and have nothing to do with dev mode.

To get dev mode you need to add the export condition on the client build to "development". `@rollup/plugin-node-resolve` has this option: https://github.com/rollup/plugins/tree/master/packages/node-resolve/#exportconditions. And then the solid-refresh plugin should be configured similar to nollup I'd guess? Unfortunately, we've been very much on Vite the last couple years and I think most other starter templates/bundlers have fallen to the wayside. And my rollup examples might be poor as they don't use dev mode, as rollup didn't always have great options.
